src/push-notifications.js

```diff
@@ -246,6 +260,24 @@ class PushNotificationsInstance {
     const options = { method: 'DELETE', path };
     await doRequest(options);
   }
+
+  async _updateDeviceMetadata() {
```

Comment:
Can we make this smarter and only update if needed? Use indexedDB to figure out what metadata has already persisted.